File: cpp/20.包含min函数的栈.cpp
class Solution {
private:
    stack<int> istk, minSt;
public:
    void push(int value) {
        istk.push(value);
        minSt.push(minSt.empty() || minSt.top() > value? value: minSt.top());
    }
    void pop() {
        istk.pop();
        minSt.pop();
    }
    int top() {
        return istk.top();
    }
    int min() {
        return minSt.top();
    }
};

File: java/20.包含min函数的栈.java
import java.util.Stack;

public class Solution {

    Stack<Integer> minStack = new Stack<Integer>();
    Stack<Integer> st = new Stack<Integer>();
    public void push(int node) {
        st.push(node);
        minStack.push(minStack.empty() || minStack.peek() > node ? node: minStack.peek());
    }
    
    public void pop() {
        st.pop();
        minStack.pop();
    }
    
    public int top() {
        return st.peek();
    }
    
    public int min() {
        return minStack.peek();
    }
}
